apps/rush-lib/src/api/Rush.ts

```diff
@@ -63,12 +103,67 @@ export class Rush {
     return PackageJsonLookup.loadOwnPackageJson(__dirname).version;
   }
 
+  private static _tryGetRushConfiguration(): RushConfiguration | undefined {
+    try {
+      if (RushConfiguration.tryFindRushJsonLocation()) {
+        return RushConfiguration.loadFromDefaultLocation({ showVerbose: true });
+      }
+    } catch (e) {
+      // ignore
```

Comment:
This seems very similar to this other code:

https://github.com/microsoft/web-build-tools/blob/3fde77c427af0e4b58f0c7899734c899b4c8475c/apps/rush-lib/src/cli/RushCommandLineParser.ts#L113-L127

If I understand right, the underlying complication is that:

1. We prefer to print the version warnings as early as possible (to minimize any malfunctions), but...
2. `_warnAboutNodeVersion()` now needs to check `rushConfiguration.suppressNodeLtsWarning`, and...
3. Loading **rush.json** is complicated (and in the "unmanged" scenario it never gets loaded at all)

Your `_tryGetRushConfiguration()` is making a quick attempt to load **rush.json** for the purposes of `_warnAboutNodeVersion()`,  and then we come back back later and redo it with the more complete **rush.json** loading.

I'm not sure I like this approach.  It seems inefficient and potentially buggy.

Based on my conversation with @mikeharder, the purpose of the `suppressNodeLtsWarning` is merely to let people use `10.0.0` instead of `10.13.0`.  It does not affect the warnings about `6.15.0` or `9.11.2` or `98.0.0` or `10.0.0-nightly20171114803cacd228`.  Thus, the incompatibilities detected by the LTS warning are relatively benign.

I suggest to:

- Move (only) the `suppressNodeLtsWarning` warning to happen later, after the proper original **rush.json** configuration has been loaded.  This should untangle the `_tryGetRushConfiguration()` mess here.
- Let's consolidate all our versions as string constants in an `@internal` API (maybe **rush-lib/api/CompatibilityVersions.ts**?).  It can also be used by **rush** (via its non-version-selector **rush-lib** dependency).  This will make it easier for us to scatter the warning checks throughout the project in best places, while still having a nice central place to keep track of them all.

Reply:
@octogonz: I'm still not clear on the precise meaning of `suppressNodeLtsWarning`.  Could you or @iclanton create a small table showing which versions of Node should get which warning(s), and which of these warnings are suppressed by the setting?

My specific concern was that Node 12 Current (e.g. `12.6.0`) would get a warning by default, and `suppressNodeLtsWarning` would be needed to suppress it.  I want to ensure our repo can use all LTS and even-numbered "Current" versions of Node with no warnings.

Reply:
Sorry about the confusion.  I used `10.0.0` as my example because it has a corresponding LTS version `10.13.0`.  The `12.x` version series doesn't have an LTS version number yet, so I was trying to avoid using it.

Let me rephrase like this:

The purpose of the `suppressNodeLtsWarning` is merely to let people use e.g. `<even>.x.x` instead of `<even>.<LTS>.x`. 

It does not affect the warnings about:
- `6.15.0` (an outdated release), or
- `9.11.2` (an `<odd>.x.x` release), or
- `98.0.0` (a "too new" release), or
- `10.0.0-nightly20171114803cacd228` (a nightly release)